baseplate/live_data/writer.py

```diff
@@ -0,0 +1,119 @@
+"""Write a file's contents to a node in ZooKeeper."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import difflib
+import logging
+import sys
+
+from kazoo.exceptions import NoNodeError, BadVersionError
+
+from .._compat import configparser
+from ..secrets import secrets_store_from_config
+from .zookeeper import zookeeper_client_from_config
+
+
+logger = logging.getLogger(__name__)
+
+
+class WriterError(Exception):
+    exit_status = 0
+
+
+class NodeDoesNotExistError(WriterError):
+    exit_status = 1
+
+    def __init__(self):
+        super(NodeDoesNotExistError, self).__init__(
+            "Target node does not exist. Please create it with appropriate "
+            "ACLs first.")
+
+
+class NoChangesError(WriterError):
+    def __init__(self):
+        super(NoChangesError, self).__init__(
+            "No changes detected. Exiting without writing.")
+
+
+class UnexpectedChangeError(WriterError):
+    exit_status = 1
+
+    def __init__(self):
+        super(UnexpectedChangeError, self).__init__(
+            "The data in ZooKeeper changed unexpectedly.")
+
+
+def write_file_to_zookeeper(zookeeper, source_file, dest_path):
+    logger.info("Writing to %s in ZooKeeper...", dest_path)
+
+    try:
+        current_data, stat = zookeeper.get(dest_path)
+        current_version = stat.version
+    except NoNodeError:
+        raise NodeDoesNotExistError
+
+    new_data = source_file.read()
```

Comment:
I don't think there's anything useful I can do with the exception once caught, so probably not.